source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
+
+By default, {+odm+} does not automatically save associated documents from
+non-embedded associations when saving the parent document. Because of this, it's
+possible to create dangling references to documents that don't exist through
+associations. You can use the ``autosave`` option on a referenced association to
+automatically save associated documents when saving the parent document. The
+following example creates a ``Band`` class with an associated ``Album`` class
+and specifies the ``autosave`` option:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-autosave
+   :end-before: # end-autosave
+   :emphasize-lines: 10
+
+.. note::
+
+   {+odm+} automatically adds autosave functionality to an association that uses
+   the ``accepts_nested_attributes_for`` option.
+
+You do not need to specify the ``autosave`` option for embedded associations
+because {+odm+} saves embedded documents in the parent document.
+
+Autobuild
+~~~~~~~~~
+
+You can add the ``autobuild`` option to one-to-one associations, such as
+``has_one`` and ``embeds_one``, to automatically instantiate a new document when
+accessing a ``nil`` association. The following example adds the ``autobuild``
+option to an association on the ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-autobuild
+   :end-before: # end-autobuild
+
+Touch
+~~~~~
+
+You can add the ``touch`` option to any ``belongs_to`` association to ensure
+that {+odm+} touches the parent document any time the child document is updated.
+When {+odm+} touches the parent document, it updates the document's
+``updated_at`` field. The following example adds the ``touch`` option to an
+association on the ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-touch
+   :end-before: # end-touch
+
+You can also specify a string or a symbol in the ``touch`` option to specify
+another field to touch on the parent association. When doing so, {+odm+} updates
+the specified field and the ``updated_at`` field in the parent association. The
+following example instructs {+odm+} to touch the ``bands_updated_at`` field:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-touch-specific
+   :end-before: # end-touch-specific
+
+.. note::
+
+   In embedded associations, when an embedded document is touched, {+odm+}
+   touches its parents recursively. Because of this, adding a ``touch``
+   attribute to an ``embedded_in`` association is unnecessary. {+odm+} does not
+   support specifying additional fields to touch in ``embedded_in``
+   associations.
+
+Counter Cache
+~~~~~~~~~~~~~
+
+You can use the ``counter_cache`` option to store the number of objects
+that belong to an associated field. When specifying this option, {+odm+} stores
```

Comment:
i: [dangling modifier](https://www.mongodb.com/docs/meta/style-guide/writing/clarify-gerunds-participles/#clarify-gerunds-and-participles) for 'specifying'

```suggestion
that belong to an associated field. When you specify this option, {+odm+} stores
```